### DISCUSSION TITLE
How to get parent's child instances?

### DISCUSSION THREAD

**Discussion by rairafaqatali** [elsa-workflows/elsa-core#1518] [Q&A]
Hi @sfmskywalker 
I have a parent workflow that triggers the child workflow. 

Is there any mechanism where I can fetch all the child instances of parent workflow?

**Comment by sfmskywalker** [comment]
Hi @rairafaqatali ,

Unfortunately there is no relationship being maintained between a parent and a child workflow.

What we could do to support this is add a new property to `WorkflowInstance` called `CustomAttributes` which would be a simple dictionary of string/object pairs.

Then the `RunWorkflow` activity could set its own instance ID as the parent ID of the child workflow instance.
To make this searchable, we would also introduce a new table called something like `WorkflowInstanceAttributes`.

This enables other scenarios as well, where you want to associate more custom data with a given workflow and search for workflow instances by a given value.

**Reply by sfmskywalker** [reply]
I created an issue for this here: https://github.com/elsa-workflows/elsa-core/issues/1524
